wrappers/kn/src/nativeMain/kotlin/zxingcpp/Barcode.kt

```diff
@@ -0,0 +1,188 @@
+			"text=$text, " +
+			"ecLevel=$ecLevel, " +
+			"symbologyIdentifier=$symbologyIdentifier, " +
+			"position=$position, " +
+			"orientation=$orientation, " +
+			"hasECI=$hasECI, " +
+			"isInverted=$isInverted, " +
+			"isMirrored=$isMirrored, " +
+			"lineCount=$lineCount" +
+			")"
+	}
+}
+
+class BarcodeImpl(
+	override val isValid: Boolean,
+	override val errorMsg: String?,
+	override val format: BarcodeFormat,
+	override val contentType: ContentType,
+	override val text: String?,
+	override val ecLevel: String?,
+	override val symbologyIdentifier: String?,
+	override val position: Position,
+	override val orientation: Int,
+	override val hasECI: Boolean,
+	override val isInverted: Boolean,
+	override val isMirrored: Boolean,
+	override val lineCount: Int,
+	private val bytes: ByteArray? = null,
+	private val bytesECI: ByteArray? = null,
+) : Barcode() {
+	override fun bytes(len: Int): ByteArray? = bytes?.sliceArray(0 until len)
+	override fun bytesECI(len: Int): ByteArray? = bytesECI?.sliceArray(0 until len)
+}
+
+@OptIn(ExperimentalForeignApi::class)
+class BoundBarcode(val cValue: CValuesRef<zxing_Barcode>) : Barcode() {
+	override val isValid: Boolean
+		get() = zxing_Barcode_isValid(cValue)
+	override val errorMsg: String?
+		get() = zxing_Barcode_errorMsg(cValue)?.toKString()
+	override val format: BarcodeFormat
+		get() = zxing_Barcode_format(cValue).parseIntoBarcodeFormat().first { it != BarcodeFormat.None }
+	override val contentType: ContentType
+		get() = zxing_Barcode_contentType(cValue).toKObject()
+
+	override fun bytes(len: Int): ByteArray? = zxing_Barcode_bytes(cValue, cValuesOf(len))?.readBytes(len)
+	override fun bytesECI(len: Int): ByteArray? = zxing_Barcode_bytesECI(cValue, cValuesOf(len))?.readBytes(len)
+	override val text: String?
+		get() = zxing_Barcode_text(cValue)?.toKString()
+	override val ecLevel: String?
+		get() = zxing_Barcode_ecLevel(cValue)?.toKString()
+	override val symbologyIdentifier: String?
+		get() = zxing_Barcode_symbologyIdentifier(cValue)?.toKString()
+	override val position: Position
+		get() = zxing_Barcode_position(cValue).useContents { toKObject() }
+	override val orientation: Int
+		get() = zxing_Barcode_orientation(cValue)
+	override val hasECI: Boolean
+		get() = zxing_Barcode_hasECI(cValue)
+	override val isInverted: Boolean
+		get() = zxing_Barcode_isInverted(cValue)
+	override val isMirrored: Boolean
+		get() = zxing_Barcode_isMirrored(cValue)
+	override val lineCount: Int
+		get() = zxing_Barcode_lineCount(cValue)
+
+	protected fun finalize() {
+		zxing_Barcode_delete(cValue)
+	}
+
+}
+
+@OptIn(ExperimentalForeignApi::class)
+fun CValuesRef<zxing_Barcode>.toKObject(): BoundBarcode = BoundBarcode(this)
+
+@OptIn(ExperimentalForeignApi::class)
+fun CValuesRef<zxing_Barcodes>.toKObject(): List<BoundBarcode> = mutableListOf<BoundBarcode>().apply {
+	for (i in 0..<zxing_Barcodes_size(this@toKObject))
+		zxing_Barcodes_at(this@toKObject, i)?.toKObject()?.let { add(it) }
```

Comment:
You need the `zxing_Barcodes_move()` function here, to move ownership of the `Barcode` object from the list (the internal std::vector) to the kotlin runtime.

Reply:
fixed in: https://github.com/zxing-cpp/zxing-cpp/pull/719/commits/fade42f851a2f3a3461d976905985a81b18ab290